src/Microsoft.AspNetCore.Mvc.Core/Authorization/AuthorizeFilter.cs

```diff
@@ -75,7 +75,6 @@ public AuthorizeFilter(AuthorizationPolicy policy)
 
             // Note: Default Anonymous User is new ClaimsPrincipal(new ClaimsIdentity())
             if (httpContext.User == null ||
-                !httpContext.User.Identities.Any(i => i.IsAuthenticated) ||
                 !await authService.AuthorizeAsync(httpContext.User, context, Policy))
             {
                 context.Result = new ChallengeResult(Policy.AuthenticationSchemes.ToArray());
```

Comment:
Not directly related to this PR, but I'm still not very comfortable with such code. It's a decision that should be made at the authorization service level, not in the authentication middleware :-1: 

@HaoK curious: do you remember the outcome of the alternate design I had suggested for `IAuthorizationService`? https://github.com/PinpointTownes/Security/commit/7ac1a2515015a98dc6f4919f28baa9db33beb50a#commitcomment-11507157

Returning an `AuthorizationResult` class/struct/enum instead of a binary bool would allow much more flexibility IMHO, and would allow replacing the current crazy design, that consists in guessing whether an `AuthenticationTicket` can be resolved by the authentication middleware to determine if a 401 or a 403 response should be returned :smile: